### ISSUE TITLE
Not responding at startup, then crashes!

### ISSUE THREAD

**Issue by play4Tutorials** [microsoft/vscode#118522]
- VS Code Version: 1.54.1 (user setup)
- Commit: f30a9b73e8ffc278e71575118b6bf568f04587c8
- Date: 2021-03-04T22:40:57.714Z(4 days ago)
- Electron: 11.3.0
- Chrome: 87.0.4280.141
- Node.js: 12.18.3
- V8: 8.7.220.31-electron.0
- OS: Windows_NT ia32 6.1.7601
- OS Version: Windows 7 SP1(Service Pack 1)

Steps to Reproduce:
1. Start VS Code

Does this issue occur when all extensions are disabled?: Don't Know

### Actual Problem Description:
This is happening since about a week or two.

When, I am starting up VSCode, it starts, but after 1 sec. the background becomes violet(my theme), then after 1 second, a popup comes saying that the window is not responding. It has 3 options. Reopen, Keep Waiting, and Close. Each explain themselves. If I click on reopen, same happens. If I click on close, VSCode closes. But if I click on keep waiting or close(X) the popup, another windows instantly pops-up, saying - the window has crashed(reason: crashed), We are sorry for your inconveniance. It has 2 options, Reopen, Close. None of them are useful. If I click reopen same thing happens, and close, closes VSCode. And if i close the popup, the window just stays there, a violet box. Previously also, this used to happen. But then I used to close the first popup. and after few seconds, VSCode used to load up as ussual. Pls fix this. And now, i have to keep on reopening for 3 to 4 times untill it comes.

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by deepak1556** [comment]
Can you also follow these steps and attach the crash report, try to replicate the scenario that resulted in ` the window has crashed(reason: crashed)` https://github.com/microsoft/vscode/wiki/Native-Crash-Issues#creating-and-symbolicating-local-crash-reports

**Comment by play4Tutorials** [comment]
I also wanted to say that when I start VSCode using the command line... Before anything.. it shows, "electron: sending uncompressed crash reports is deprecated. crash reports will be gzipped and sent.".... i don't know, if these are related, just wanted to let you know.

> Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

definitely...

> Can you also follow these steps and attach the crash report, try to replicate the scenario that resulted in ` the window has crashed(reason: crashed)` https://github.com/microsoft/vscode/wiki/Native-Crash-Issues#creating-and-symbolicating-local-crash-reports

for that i need to restart, i'll try that tomorrow.

**Comment by bpasero** [comment]
Thanks

**Comment by deepak1556** [comment]
> Before anything.. it shows, "electron: sending uncompressed crash reports is deprecated. crash reports will be gzipped and sent."

This should not be seen if you are running `1.54.1`, it is fixed in this version.

**Comment by play4Tutorials** [comment]
> Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

Yeah, it doesn't occur in Insiders...

**Comment by play4Tutorials** [comment]
> Can you also follow these steps and attach the crash report, try to replicate the scenario that resulted in ` the window has crashed(reason: crashed)` https://github.com/microsoft/vscode/wiki/Native-Crash-Issues#creating-and-symbolicating-local-crash-reports

So, I followed the steps, but when again today it happened, I checked the directory and `*.dmp` was generated...

**Comment by play4Tutorials** [comment]
^^^^^^^^^^^^^^^^
sorry, I meant **no** `*.dmp` file was generated.

**Comment by deepak1556** [comment]
Can you share a screen recording of the steps you tried for https://github.com/microsoft/vscode/issues/118522#issuecomment-798051638

**Comment by play4Tutorials** [comment]
sorry, I have a little slow pc, and obs and vs code together just almost freeze my pc, i could provide a few screenshots, though.... would that do?

**Comment by play4Tutorials** [comment]
BTW, could you please have a look at #118705. I opened it almost 3 days ago, and no one has said anything about it??!!? 
@deepak1556

**Comment by appurist** [comment]
As far as I can tell, I'm having the same problem. But to clarify, it is not what I would call a crash, it is a hang. The process does not exit, but interaction with the app causes the system to jump in with a "not responding" dialog where you can force-kill it.

This is a show-stopper problem for me as I cannot use VS Code _at all_ anymore.

I have downloaded and installed the insiders nightly version. I have followed the instructions on https://github.com/microsoft/vscode/wiki/Native-Crash-Issues#creating-and-symbolicating-local-crash-reports but it does not seem very helpful as only the clientid files are generated (because the app does not actually crash).

I have tried waiting longer, but that doesn't seem to help. Symptoms are that you cannot close the welcome page, you cannot uncheck the "show welcome" checkbox, the window close box does nothing, i.e. the application does not respond to events.

An attempt to resize the window does resize the frame (see image below) but does not resize the contents, as the app is frozen hard. (See whitespace on right and bottom of window in image below.)

A short after attempting to interact, the system displays a "not responding" dialog and offers a forced quit.

![vscode1](https://user-images.githubusercontent.com/1620487/111081365-b854b800-84e1-11eb-9214-6f7785c62f13.png)

This is the latest nightly insider as of Sunday March 14th 3pm, I used `code-insiders-1.55.0-1615526749.el7.x86_64.rpm` to install, and created an empty folder and used the following command to launch VS Code:
`code-insiders --crash-reporter-directory ~/crashes .`

The contents of the `~/crashes` folder only contains a `client_id` file with an ID in it:
`69c31889-2c99-4fed-94e2-f3275f581566`
and a folder named `exthost Crash Reports` with another `client_id` file with an ID in it:
`17b573a4-9c8a-47e0-9c6a-11c680d6e828`
Note they are different IDs in spite of being named `client_id`, not sure if that is expected.

The machine is a CentOS 7 installation with the latest Chrome installed (89 I believe) and it runs fine (I'm using it to submit this report).

If there is a file-based way to disable the welcome screen before starting code, I could do that too, to determine if it is something related to rendering that page.  Also if there is a file-based method to disable all extensions, I could try that.

**Comment by appurist** [comment]
I just tried removing the Code-related folders under ~/.config but I still get the same hang.

I checked for any output reports and found the `logs` subfolder, with two dated subfolders under that. One of them was empty, the other included two files, `extensions.log` and `tasks.log`, but both of those files were 0 bytes long.  Force-killing VS Code (in case of buffered writes, pending a close) did not add any bytes to them.

**Comment by play4Tutorials** [comment]
@appurist Your image shows a complete hang, wheras that is not what happens with me. My one loads, but after opening and closing it three to four times(sometimes 2 or maybe 5).... But yours is just frozen, and as you said I would also not call this a crash....

**Comment by appurist** [comment]
@play4Tutorials I'll open a separate issue for the hang case. (It may be related, or not, but that is for the Code developers to determine.)

**Comment by deepak1556** [comment]
> sorry, I have a little slow pc, and obs and vs code together just almost freeze my pc, i could provide a few screenshots, though.... would that do?

I was mainly interested to see if you closed the app after the `the window has crashed(reason: crashed)` dialog which would generate the crash report, if you had closed the app while the unresponsive dialog was present it would not generate the report.

If you are unable to get the reports, please follow the below steps as an alternative way to get the memory dump while the window is unresponsive.

* Download and install latest vscode insiders https://code.visualstudio.com/insiders
* Download https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer
* Find the workbench process id by following command in powershell
```
Get-WmiObject win32_process -Filter "name='Code - Insiders.exe'" | where CommandLine -like '*--type=renderer*' | Select-Object ProcessId, CommandLine
```
* Once the app freezes, open process explorer and find the processes with above ProcessId
* Right Click on the Process -> Create Dump -> Full Dump
* Send the files to Deepak.Mohan@microsoft.com

![Screenshot (11)](https://user-images.githubusercontent.com/964386/111221389-aa616e80-8597-11eb-936d-423e6c0dbd02.png)

**Comment by play4Tutorials** [comment]
@deepak1556 Sure I would try the above steps to generate the dump.....

And could I attach the dump file here???

And I closed the window on the crashed dialog, yet nothing was generated.

**Comment by play4Tutorials** [comment]
It gives the following error(the powershell command)

BTW, I am using Powershell, v2.0, Copyright 2009

Where-Object : Cannot bind parameter 'FilterScript'. Cannot convert the "CommandLine" value of type "System.String" to type "System.Management.Automa
tion.ScriptBlock".
At line:1 char:62
+ Get-WmiObject win32_process -Filter "name='Code.exe'" | where <<<<  CommandLine -like '*--type=renderer*' | Select-Object ProcessId
    + CategoryInfo          : InvalidArgument: (:) [Where-Object], ParameterBindingException
    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,Microsoft.PowerShell.Commands.WhereObjectCommand

**Comment by play4Tutorials** [comment]
> @play4Tutorials I'll open a separate issue for the hang case. (It may be related, or not, but that is for the Code developers to determine.)

true @appurist

**Comment by appurist** [comment]
For the record, it turns out my issue was completely unrelated to this one. Mine is Linux-specific; Microsoft dropped support for CentOS 7 and RHEL 7 (the only long-term support releases available to enterprises) in a recent minor update. Worse, the error handling for this change is horrible -- the app just hangs hard -- so it took a while to figure this out.  It cannot possibly be the issue reported in this thread. Sorry for the distraction.

**Comment by deepak1556** [comment]
> And could I attach the dump file here???

Please send the files to Deepak.Mohan@microsoft.com

> BTW, I would be attaching for the non-insiders version

Try the insiders version, the insiders version gets promoted to next stable and if the issue is present then generate the logs for this version.

> It gives the following error(the powershell command)

Try with a newer version of powershell

**Comment by play4Tutorials** [comment]
> Please send the files to Deepak.Mohan@microsoft.com

Alright!

> Try the insiders version, the insiders version gets promoted to next stable and if the issue is present then generate the logs for this version.

How do you generate logs???

> Try with a newer version of powershell

I can't.... I don't have a newer version.... sorry... is there any command for cmd or powershell v2.0...??

**Comment by play4Tutorials** [comment]
Also, yesterday my VS Code updated to 1.54.3, and now, keep waiting kind of resolves the issue, because when I click on it, it kind of freezes, not actually but visually, and after 1 min, it loads as usual...

**Comment by play4Tutorials** [comment]
> For the record, it turns out my issue was completely unrelated to this one. Mine is Linux-specific; Microsoft dropped support for CentOS 7 and RHEL 7 (the only long-term support releases available to enterprises) in a recent minor update. Worse, the error handling for this change is horrible -- the app just hangs hard -- so it took a while to figure this out. It cannot possibly be the issue reported in this thread. Sorry for the distraction.

Yeah, sorry to hear that, maybe move to another distro, as ubuntu, since as far as I know, are pretty popular, or debian, maybe.. I have used ubuntu and kali linux myself, but none other.. seen them in action though... so sorry... @appurist

**Comment by deepak1556** [comment]
> How do you generate logs???

I meant getting logs based on the steps in https://github.com/microsoft/vscode/issues/118522#issuecomment-799752615. Ignore the powershell step in there, it was just an easy way to find the process id. You can just open the process explorer and get the memory dump from `Code - Insiders.exe` process that includes the command line value `--type=renderer`

**Comment by play4Tutorials** [comment]
Thanks for all of your help... It's really appreciated, but I am closing this issue since it has been fixed in the latest update... Thanks to the whole VS Code team, keep improving VS Code.. Thanks, again...
